### PR TITLE
Fix allRegistries functionality and add error handling for API requests

### PR DESCRIPTION
**Pull request by kusalk**
Unfortunately #184 broke support for private/custom package repositories. This rectifies.

### CODE REVIEW

File: src/fixLockfileIntegrity.ts [hunk 1]
```diff
@@ -26,37 +26,57 @@ const prettierInitialConfig: prettier.Options = {
...
     };
-    const metadata: GotResponse<NPMJS_METADATA_PARTIAL> = await got.get<NPMJS_METADATA_PARTIAL>(url, options);
+    let metadata: Response<NPMJS_METADATA_PARTIAL>;
+    try {
+        metadata = await got.get<NPMJS_METADATA_PARTIAL>(url, options);
+    }
+    catch (error) {
+        logger.warn(`${chalk.red("Error retrieving response from API:")} ${chalk.blue(url.toString())} - ${chalk.red(error.message)}`);
+        return undefined;
+    }
```
Comment (yoavain) [inline src/fixLockfileIntegrity.ts:57]: I saw you added a test that mocks rejection by `got`, but did you see this in a real flow? 
Can you simulate an example where `throwHttpErrors: false` is not enough and `got` still throws? (no need to add a test, an example URL is enough, I'll test locally)

Comment (kusalk) [reply src/fixLockfileIntegrity.ts:57]: The error I was seeing was `Error: getaddrinfo ENOTFOUND` and the whole script would crash. So for any domain that doesn't resolve.

Comment (yoavain) [reply src/fixLockfileIntegrity.ts:57]: Got it!
(pun intended)

Comment (kusalk) [reply src/fixLockfileIntegrity.ts:57]: Let me double check this on my end

Comment (kusalk) [reply src/fixLockfileIntegrity.ts:57]: Can confirm it catches the error as expected. However I noticed another unrelated bug - I'll create a separate issue for that :)

Comment (kusalk) [reply src/fixLockfileIntegrity.ts:57]: Okay created an issue for that https://github.com/yoavain/fix-lockfile-integrity/issues/235 - I think we don't have to worry about that scenario very much as it is rare.

**Review by yoavain — COMMENTED** [review]
Looks good

**Pull request merged**